File: AStarCSharp/AStar/astar/graphics/PrintMap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AStar.astar.graphics
{
    class PrintMap
    {
        public PrintMap(AStarMap map, List<Point> shortestPath) 
        {
            StringBuilder sb = new StringBuilder();
		    AStarCell cell;
		    for(int y = 0; y < map.getMapHeight(); y++) 
            {
			    if(y == 0) 
                {
				    for (int i = 0; i <= map.getMapHeight(); i++)
                    {
                        sb.Append("-");
                    }
				    sb.AppendLine();	
			    }
			    sb.Append("|");

			    for(int x = 0; x < map.getMapWith(); x++) 
                {
				    cell = map.getCell(x, y);
				
				    if(cell == null || cell.isObstacle()) 
                    {
					    sb.Append("X");
				    } 
                    else if(isStart(shortestPath, cell.getPoint())) 
                    {
					    sb.Append("s");
				    } 
                    else if(isGoal(shortestPath, cell.getPoint())) 
                    {
					    sb.Append("g");
				    } 
                    else if (contains(shortestPath, cell.getPoint())) 
                    {
					    sb.Append("?");
				    } 
                    else 
                    {
					    sb.Append(" ");
				    }
				    if(y==map.getMapHeight())
                    {
                        sb.Append("_");
                    }
			    }

			    sb.Append("|");
			    sb.AppendLine();
		    }
		    for (int i = 0; i <= map.getMapHeight(); i++)
            {
                sb.Append("-");
            }
            System.Console.SetWindowSize(System.Console.LargestWindowWidth, System.Console.LargestWindowHeight);
            System.Console.Write(sb.ToString());
	    }

        private static bool contains(List<Point> shortestPath, Point point)
        {
            foreach (Point p in shortestPath)
            {
                if (p.Equals(point))
                    return true;
            }

            return false;
        }

        private static bool isStart(List<Point> shortestPath, Point point)
        {
            if (shortestPath.Count <= 0)
            {
                return false;
            }

            return shortestPath[0].Equals(point);
        }

        private static bool isGoal(List<Point> shortestPath, Point point)
        {
            if (shortestPath.Count <= 0)
            {
                return false;
            }

            return shortestPath[shortestPath.Count - 1].Equals(point);
        }
    }

}

File: AStarJava/src/pathFinding/graphics/PrintMap.java
package pathFinding.graphics;

import java.awt.Point;
import java.util.ArrayList;

import pathFinding.core.AStarCell;
import pathFinding.core.AStarMap;

public class PrintMap {
	public PrintMap(AStarMap map, ArrayList<Point> shortestPath) {
		AStarCell cell;
		for(int y = 0; y < map.getHeightInCells(); y++) {
			if(y == 0) {
				for (int i = 0; i <= map.getHeightInCells(); i++)
					System.out.print("-");
				System.out.println();	
			}
			System.out.print("|");

			for(int x = 0; x < map.getWidthInCells(); x++) {
				cell = map.getCell(x, y);
				if(AStarCell.isObstacle(cell)) {
					System.out.print("X");
				} else if(isStart(shortestPath, cell.getPoint())) {
					System.out.print("s");
				} else if(isGoal(shortestPath, cell.getPoint())) {
					System.out.print("g");
				} else if (shortestPath.contains(new Point(cell.getX(), cell.getY()))) {
					System.out.print("?");
				} else {
					System.out.print(" ");
				}
				if(y == map.getHeightInCells())
					System.out.print("_");
			}

			System.out.print("|");
			System.out.println();
		}
		for (int i = 0; i <= map.getHeightInCells(); i++)
			System.out.print("-");
	}
	
	private static boolean isStart(ArrayList<Point> shortestPath, Point point) {
		if(shortestPath.isEmpty()) {
			return false;
		}
		
		return shortestPath.get(0).equals(point);
	}
	
	private static boolean isGoal(ArrayList<Point> shortestPath, Point point) {
		if(shortestPath.isEmpty()) {
			return false;
		}
		
		return shortestPath.get(shortestPath.size() - 1).equals(point);
	}
}
